run_pacta_data_preparation.R

```diff
@@ -504,20 +504,38 @@ iss_company_emissions <-
 
 logger::log_info("Formatting and saving file: \"iss_entity_emission_intensities.rds\".")
 
+factset_financial_data <-
+  readRDS(factset_financial_data_path) %>%
+  select(factset_entity_id,issue_type, adj_price, adj_shares_outstanding) %>%
+  filter(!is.na(factset_entity_id) & !is.na(adj_shares_outstanding)) %>%
+  filter(issue_type %in% c("EQ", "PF", "CP")) %>%
+  summarize(
+    mkt_val = sum(adj_price * adj_shares_outstanding, na.rm = TRUE),
+    .by = "factset_entity_id"
+  )
+
+factset_entity_info <-
+  readRDS(factset_entity_info_path) %>%
+  select(factset_entity_id, iso_country, sector_code, factset_sector_desc) %>%
+  left_join(factset_financial_data, by = "factset_entity_id")
+
+rm(factset_financial_data)
+invisible(gc())
+
 iss_entity_emission_intensities <-
   readRDS(factset_entity_financing_data_path) %>%
+  left_join(factset_entity_info, by = "factset_entity_id") %>%
+  filter(countrycode::countrycode(iso_country, "iso2c", "iso4217c") == currency) %>%
   left_join(currencies, by = "currency") %>%
   mutate(
-    ff_mkt_val = ff_mkt_val * exchange_rate,
-    ff_debt = ff_debt * exchange_rate,
+    ff_mkt_val = if_else(!is.na(mkt_val), mkt_val, NA_real_),
+    ff_debt = if_else(!is.na(ff_debt), ff_debt * exchange_rate, NA_real_),
```

Comment:
Does `mkt_val` not need to be multiplied by `exchange_rate` as well?

Reply:
better answered by @Antoine-Lalechere

Reply:
Here mkt_val comes from factset_financial_data extracted from Factset Ownership bundle and all Factset Ownership data is already in USD.